src/esp/physics/bullet/BulletRigidObject.cpp

```diff
@@ -459,29 +456,30 @@ void BulletRigidObject::activateCollisionIsland() {
   // first query overlapping pairs of the current object from the most recent
   // broadphase to collect relevant simulation islands.
 
-  std::set<int> overlappingSimulationIslands = {thisColObj->getIslandTag()};
+  // each index represents an island tag present - default is -1, so add one.
+  std::bitset<65536> overlappingSimIslands;
+  overlappingSimIslands[thisColObj->getIslandTag() + 1] = true;
   auto* bColWorld = bWorld_->getCollisionWorld();
   auto& pairCache = bColWorld->getPairCache()->getOverlappingPairArray();
   for (int i = 0; i < pairCache.size(); ++i) {
     if (pairCache.at(i).m_pProxy0->m_clientObject == thisColObj) {
-      overlappingSimulationIslands.insert(
-          static_cast<btCollisionObject*>(
-              pairCache.at(i).m_pProxy1->m_clientObject)
-              ->getIslandTag());
+      overlappingSimIslands[static_cast<btCollisionObject*>(
+                                pairCache.at(i).m_pProxy1->m_clientObject)
+                                ->getIslandTag() +
+                            1] = true;
     } else if (pairCache.at(i).m_pProxy1->m_clientObject == thisColObj) {
-      overlappingSimulationIslands.insert(
-          static_cast<btCollisionObject*>(
-              pairCache.at(i).m_pProxy0->m_clientObject)
-              ->getIslandTag());
+      overlappingSimIslands[static_cast<btCollisionObject*>(
+                                pairCache.at(i).m_pProxy0->m_clientObject)
+                                ->getIslandTag() +
+                            1] = true;
```

Comment:
My obsession with not using unnecessarily heavy APIs probably shows here, but anyway -- using `.set()` might be nicer to the compiler (and debug build speed) as it doesn't have to create a temporary "reference" object that performs the bit assignment.

The `[]` query below is fine, tho.

Reply:
Tried set first, apparently it will potentially throw an exception if out of range 👯

Reply:
[] just pukes

Reply:
ugh STL

alternatively you could use my [BoolVector](https://doc.magnum.graphics/magnum/classMagnum_1_1Math_1_1BoolVector.html), that one isn't riddled with exceptions :))